lib/hyacinth/adapters/preservation_adapter/fedora3/text_resource_properties.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Hyacinth
+  module Adapters
+    module PreservationAdapter
+      class Fedora3::TextResourceProperties
+        include Fedora3::DatastreamMethods
+
+        def self.from(hyacinth_obj)
+          new(hyacinth_obj)
+        end
+
+        def initialize(hyacinth_obj)
+          @hyacinth_obj = hyacinth_obj
+        end
+
+        def to(fedora_obj)
+          return unless @hyacinth_obj.is_a? ::DigitalObject::Asset
+          ::DigitalObject::Asset::TEXT_RESOURCE_NAMES.each do |resource_name|
+            next unless @hyacinth_obj.send :"has_#{resource_name}_resource?"
+            resource = @hyacinth_obj.send :"#{resource_name}_resource"
+            resource_config = @hyacinth_obj.class.resource_attributes.fetch(resource_name.to_sym, {})
+            next unless resource_config[:preservable] && resource.content_exists?
+            ensure_datastream(fedora_obj, resource_name, datastream_props_for(resource_name, resource_config, resource))
+            datastream = fedora_obj.datastreams[resource_name]
+            # Only update content if it has changed
+            next if "sha256:#{datastream.checksum}" == resource.checksum
```

Comment:
I remember you mentioning checksums on text resources vs larger ones on non-text resources.  This is making me think that we want to add a `verify_checksum: true|false` config option to resource_attributes as well, and only have fedora verify the checksum if that's marked as true (but always store the checksum as a RELS-INT property if present, regardless of verify_checksum property value).

Reply:
it also only makes sense for FCR-managed content, really

Reply:
True.  I'll resolve this conversation since we talked about nesting the arguments under the `preserve` key.